src/java.desktop/share/classes/javax/swing/JEditorPane.java

```diff
@@ -619,35 +619,37 @@ void read(InputStream in, Document doc) throws IOException {
         String charset = (String) getClientProperty("charset");
         try(Reader r = (charset != null) ? new InputStreamReader(in, charset) :
                 new InputStreamReader(in)) {
```

Comment:
The changeset looks confusing in the diff. It becomes clearer if you disable showing whitespace differences.

You could've elaborated on the fix in the description.

The fix is to add a nested `try`-block inside `try-with-resource`; all the exceptions are handled in the nested `try`; the outer `try`-with-resouces only closes the input stream.

Reply:
Yeah, I too agree on this point. The fix description has to elaborated, we are not able to make out in diff.

Reply:
I pulled the change into my local workspace to test it. IDE hides white-space differences and the fix becomes clear. I hid white-space differences in GitHub diff viewer: only two lines are changed. :)